app/code/Magento/Backend/Block/Dashboard/Orders/Grid.php

```diff
@@ -92,7 +92,7 @@ protected function _prepareCollection()
     protected function _afterLoadCollection()
     {
         foreach ($this->getCollection() as $item) {
-            $item->getCustomer() ?: $item->setCustomer('Guest');
+            $item->getCustomer() ?: $item->setCustomer($item->getBillingAddress()->getName());
```

Comment:
What is the reason to get customer name from the Billing Address?

Reply:
@ishakhsuvarov, as it is guest, so he has only shipping and billing addresses. I think it is more reasonably to take it from billing than from shipping. 
Also I checked the mail message after I had placed order as guest. It also takes name from billing address.